tools/autograd/gen_variable_type.py

```diff
@@ -258,9 +297,31 @@ def get_type(arg_name):
                         sizes_name = arg['name'] + '_sizes'
                         formula = formula.replace(size_str, sizes_name)
 
+                    # If x is a TensorList, turn x.sizes(y) into x_argsizes_y
+                    def argsizes_repl(matchobj):
+                        if arg['type'] != 'TensorList':
+                            raise RuntimeError("sizes(argument) only supported on TensorList")
+                        argsizes_name = arg['name'] + "_argsizes_" + matchobj.group(1)
+                        arg_sizes_found.append(argsizes_name + ".size()")
+                        return argsizes_name
+                    formula = re.sub(arg['name'] + r".sizes\((\w+)\)", argsizes_repl, formula)
+
+                    # If x is a Tensor, turn x.size(y) into x_argsize_y
+                    def argsize_repl(matchobj):
+                        if arg['type'] != 'Tensor':
+                            raise RuntimeError("size(argument) only supported on Tensor")
+                        argsize_name = arg['name'] + "_argsize_" + matchobj.group(1)
+                        return argsize_name
+                    formula = re.sub(arg['name'] + r".size\((\w+)\)", argsize_repl, formula)
+
                 derivatives.append(formula)
                 arg['derivative'] = formula
-                option['num_inputs'] += 1
+                if arg['type'] != "TensorList":
+                    option['num_inputs'] += 1
+
+        if arg_sizes_found:
+            option['num_inputs'] = ("+".join(arg_sizes_found) +
+                                    "" if option['num_inputs'] == 0 else " + " + str(option['num_inputs']))
```

Comment:
This is weird. It seems that 4 lines above `option['num_inputs']` is an int, but here you assign a string to it?

Reply:
it is weird, but it's just dumped to code at the end anyway.  I'll just separate out the integer and string variables.

Reply:
I know. I just can imagine myself doing `if option['num_inputs'] == 5:` in the code and it will silently break because of this